src/calculate_bucket.hpp

```diff
@@ -276,11 +276,14 @@ class FxCalculator {
     // bucket length, we can store all the R values and lookup each of our 32 candidates to see if
     // any R value matches. This function can be further optimized by removing the inner loop, and
     // being more careful with memory allocation.
-    inline std::vector<std::pair<uint16_t, uint16_t>> FindMatches(
+    inline void FindMatches(
         const std::vector<PlotEntry>& bucket_L,
-        const std::vector<PlotEntry>& bucket_R)
+        const std::vector<PlotEntry>& bucket_R,
+        uint16_t *idx_L,
+        uint16_t *idx_R,
+        uint32_t &idx_count)
```

Comment:
if we want to squeeze the last bit of throughput out of this, it may be better to take this by value and return the new index. That way the counter can live in a register and not have to spill to RAM

Reply:
done